lib/ansible/modules/cloud/opennebula/one_vm_info.py

```diff
@@ -0,0 +1,448 @@
+LCM_STATES = ['LCM_INIT', 'PROLOG', 'BOOT', 'RUNNING', 'MIGRATE', 'SAVE_STOP',
+              'SAVE_SUSPEND', 'SAVE_MIGRATE', 'PROLOG_MIGRATE', 'PROLOG_RESUME',
+              'EPILOG_STOP', 'EPILOG', 'SHUTDOWN', 'STATE13', 'STATE14', 'CLEANUP_RESUBMIT', 'UNKNOWN', 'HOTPLUG', 'SHUTDOWN_POWEROFF',
+              'BOOT_UNKNOWN', 'BOOT_POWEROFF', 'BOOT_SUSPENDED', 'BOOT_STOPPED', 'CLEANUP_DELETE', 'HOTPLUG_SNAPSHOT', 'HOTPLUG_NIC',
+              'HOTPLUG_SAVEAS', 'HOTPLUG_SAVEAS_POWEROFF', 'HOTPULG_SAVEAS_SUSPENDED', 'SHUTDOWN_UNDEPLOY']
+
+
+def get_vm_info(client, vm):
+    # get additional info
+    vm = client.vm.info(vm.ID)
+    # get Uptime
+    current_time = time.localtime()
+    vm_start_time = time.localtime(vm.STIME)
+    vm_uptime = time.mktime(current_time) - time.mktime(vm_start_time)
+    vm_uptime /= (60 * 60)
+
+    vm_labels, vm_attributes = get_vm_labels_and_attributes_dict(client, vm.ID)
+
+    # get disk info
+    disks = []
+    if 'DISK' in vm.TEMPLATE:
+        if isinstance(vm.TEMPLATE['DISK'], list):
+            for disk in vm.TEMPLATE['DISK']:
+                disk_entry = dict()
+                disk_entry['SIZE'] = disk['SIZE'] + ' MB'
+                disk_entry['DISK_ID'] = disk['DISK_ID']
+                disks.append(disk_entry)
+        else:
+            disk_entry = dict()
+            disk_entry['SIZE'] = vm.TEMPLATE['DISK']['SIZE'] + ' MB'
+            disk_entry['DISK_ID'] = vm.TEMPLATE['DISK']['DISK_ID']
+            disks.append(disk_entry)
+
+    # get network info
+    networks_info = []
+    if 'NIC' in vm.TEMPLATE:
+        if isinstance(vm.TEMPLATE['NIC'], list):
+            for nic in vm.TEMPLATE['NIC']:
+                networks_info.append({'ip': nic['IP'], 'mac': nic['MAC'], 'name': nic['NETWORK'], 'security_groups': nic['SECURITY_GROUPS']})
+        else:
+            networks_info.append(
+                {'ip': vm.TEMPLATE['NIC']['IP'], 'mac': vm.TEMPLATE['NIC']['MAC'],
+                    'name': vm.TEMPLATE['NIC']['NETWORK'], 'security_groups': vm.TEMPLATE['NIC']['SECURITY_GROUPS']})
+
+    info = {
+        'id': vm.ID,
+        'name': vm.NAME,
+        'state': VM_STATES[vm.STATE],
+        'user_name': vm.UNAME,
+        'user_id': vm.UID,
+        'group_name': vm.GNAME,
+        'group_id': vm.GID,
+        'lcm_state': LCM_STATES[vm.LCM_STATE],
+        'uptime_h': int(vm_uptime),
+        'mode': parse_vm_permissions(client, vm),
+        'cpu': vm.TEMPLATE['CPU'],
+        'vcpu': vm.TEMPLATE['VCPU'],
+        'memory': vm.TEMPLATE['MEMORY'] + ' MB',
+        'disks': disks,
+        'networks': networks_info,
+        'labels': vm_labels,
+        'attributes': vm_attributes
+    }
+    return info
+
+
+def get_vms_by_ids(module, client, ids):
+    vms = []
+    pool = get_all_vms(client)
+
+    for vm in pool.VM:
+        if str(vm.ID) in ids:
```

Comment:
I suggest removing the conversion to str otherwise, the ids in the playbook needs to be str as well.

Reply:
I was a little puzzelt as i found out that this
```
    - one_vm_info:
        id: 425
```
resultet in a list with one item which is a string and that this 
```
    - one_vm_info:
        id:
          - 425
          - 426
```
resultet in a list with two items which is are int.
i check for this and removed the conversion.